### PR TITLE
Adds wake up function to garrison module

### PR DESCRIPTION
**Pull request by headswe**
[Video](https://www.youtube.com/watch?v=AYksyN9hPlI)

### CODE REVIEW

File: addons/ai/functions/fnc_garrisonQuantity.sqf [hunk 1]
```diff
@@ -118,6 +126,25 @@ for "_i" from 1 to (_aiNumberToSpawn min _freeBuildingSpaces) do {
...
+        _buildingCenter set [2,0]; // set to ground
+        private _buildingSize = (boundingBox _building) # 2;
+        private _trigger = createTrigger ["EmptyDetector", _buildingCenter, false];
+        _trigger setVariable ['side', _side];
+        _trigger setVariable ['units',_building getVariable [QGVAR(garrisonedUnits), []]];
+        _trigger setTriggerInterval 2; // set interval to 2 seconds
+        _trigger setTriggerArea [_buildingSize * 3, _buildingSize * 3, 0, false, 10];
+        _trigger setTriggerActivation ["ANYPLAYER","PRESENT",false];
+        _trigger setTriggerStatements [
+            QUOTE(private _side = (thisTrigger getVariable [ARR_2('side', opfor)]); this && ({ side _x != _side} count thisList) > 0),
```
Comment (Freddo3000) [inline addons/ai/functions/fnc_garrisonQuantity.sqf:142]: Is QUOTE() needed here seeing as you're not using GVAR() or other macros?

File: addons/ai/functions/fnc_garrisonQuantity.sqf [hunk 2]
```diff
@@ -118,6 +126,25 @@ for "_i" from 1 to (_aiNumberToSpawn min _freeBuildingSpaces) do {
...
+        _buildingCenter set [2,0]; // set to ground
+        private _buildingSize = (boundingBox _building) # 2;
+        private _trigger = createTrigger ["EmptyDetector", _buildingCenter, false];
+        _trigger setVariable [QGVAR(side), _side];
+        _trigger setVariable [QGVAR(units), _building getVariable [QGVAR(garrisonedUnits), []]];
+        _trigger setTriggerInterval 2; // set interval to 2 seconds
+        _trigger setTriggerArea [_buildingSize * 3, _buildingSize * 3, 0, false, 10];
+        _trigger setTriggerActivation ["ANYPLAYER","PRESENT",false];
+        _trigger setTriggerStatements [
+            QUOTE(private _side = (thisTrigger getVariable [ARR_2(QGVAR(side), opfor)]); this && ({ side _x != _side} count thisList) > 0),
```
Comment (Freddo3000) [inline addons/ai/functions/fnc_garrisonQuantity.sqf:142]: QQGVAR()

File: addons/ai/functions/fnc_garrisonQuantity.sqf [hunk 3]
```diff
@@ -118,6 +126,25 @@ for "_i" from 1 to (_aiNumberToSpawn min _freeBuildingSpaces) do {
...
+            QUOTE({ _x enableAI 'PATH'; _x setUnitPos 'AUTO'; } forEach (thisTrigger getVariable [ARR_2(QGVAR(units), [])]);),
```
Comment (Freddo3000) [inline addons/ai/functions/fnc_garrisonQuantity.sqf:143]: QQGVAR()

Comment (headswe) [reply addons/ai/functions/fnc_garrisonQuantity.sqf]: Thanks

File: addons/ai/functions/fnc_garrisonQuantity.sqf [hunk 4]
```diff
@@ -118,6 +126,25 @@ for "_i" from 1 to (_aiNumberToSpawn min _freeBuildingSpaces) do {
...
+        private _trigger = createTrigger ["EmptyDetector", _buildingCenter, false];
+        _trigger setVariable [QGVAR(side), _side];
+        _trigger setVariable [QGVAR(units), _building getVariable [QGVAR(garrisonedUnits), []]];
+        _trigger setTriggerInterval 2; // set interval to 2 seconds
+        _trigger setTriggerArea [_buildingSize * 3, _buildingSize * 3, 0, false, 10];
+        _trigger setTriggerActivation ["ANYPLAYER","PRESENT",false];
+        _trigger setTriggerStatements [
+            QUOTE(private _side = (thisTrigger getVariable [ARR_2(QQGVAR(side), opfor)]); this && ({ side _x != _side} count thisList) > 0),
+            QUOTE({ _x enableAI 'PATH'; _x setUnitPos 'AUTO'; } forEach (thisTrigger getVariable [ARR_2(QQGVAR(units), [])]);),
+            ""
```
Comment (Freddo3000) [inline addons/ai/functions/fnc_garrisonQuantity.sqf:144]: Perhaps set it to delete itself after executing?

Comment (headswe) [reply addons/ai/functions/fnc_garrisonQuantity.sqf:144]: it only executes once and then it's defunct so.

File: addons/ai/functions/fnc_garrisonQuantity.sqf (hunk 1 above)
Comment (Freddo3000) [reply addons/ai/functions/fnc_garrisonQuantity.sqf:142]: Now it isn't needed again

File: addons/ai/functions/fnc_garrisonQuantity.sqf [hunk 5]
```diff
@@ -118,6 +126,25 @@ for "_i" from 1 to (_aiNumberToSpawn min _freeBuildingSpaces) do {
...
+        _buildingCenter set [2,0]; // set to ground
+        private _buildingSize = (boundingBox _building) # 2;
+        private _trigger = createTrigger ["EmptyDetector", _buildingCenter, false];
+        _trigger setVariable ["side", _side];
+        _trigger setVariable ["units", _building getVariable [QGVAR(garrisonedUnits), []]];
+        _trigger setTriggerInterval 2; // set interval to 2 seconds
+        _trigger setTriggerArea [_buildingSize * 3, _buildingSize * 3, 0, false, 10];
+        _trigger setTriggerActivation ["ANYPLAYER","PRESENT",false];
+        _trigger setTriggerStatements [
+            QUOTE(private _side = (thisTrigger getVariable [ARR_2('side', opfor)]); this && ({ side _x != _side} count thisList) > 0),
```
Comment (Freddo3000) [inline addons/ai/functions/fnc_garrisonQuantity.sqf:142]: Now it isn't needed again

Comment (headswe) [reply addons/ai/functions/fnc_garrisonQuantity.sqf:142]: Had issues with QQGVAR, so just went back, QUOTE makes the code lintable in my editor so it stays.

**Pull request merged**